fusion-cli/build/babel-plugins/babel-plugin-i18n/index.js

```diff
@@ -45,6 +49,55 @@ function i18nPlugin(babel /*: Object */, {translationIds} /*: PluginOpts */) {
             }
             translationIds.add(element.value);
           });
+        } else if (specifierName === 'useTranslations') {
+          if (!t.isVariableDeclarator(refPath.parentPath.parent)) {
+            throw new Error(
+              'Unexpected assignment of useTranslations return function'
+            );
+          }
+          const localName = refPath.parentPath.parent.id.name;
+          const translationPaths =
+            refPath.parentPath.scope.bindings[localName].referencePaths;
+          translationPaths.forEach(translationPath => {
+            if (
+              t.isCallExpression(translationPath.parentPath) &&
+              translationPath.parentKey === 'callee'
+            ) {
+              const arg = translationPath.parentPath.node.arguments[0];
+              const errorMessage =
+                'useTranslations result function must be passed string literal or hinted template literal';
+              if (t.isStringLiteral(arg)) {
+                translationIds.add(arg.value);
+              } else if (t.isTemplateLiteral(arg)) {
+                const result = arg.quasis
+                  .map(q => q.value.cooked)
+                  .filter((str, index) => {
+                    if (
+                      index !== 0 &&
+                      index !== arg.quasis.length - 1 &&
+                      str === ''
+                    ) {
+                      // interpolations in the middle of a string can be ignored
+                      // there is no guarantee that they will evaluate to a string of any length
```

Comment:
I think I'm a bit confused by this comment.

Regarding the first part:
>  interpolations in the middle of a string can be ignored

My reading of this statement implies the substitution in (the middle of) `` `hello${interpolation}world` `` could be ignored. But in this case, the substitution seems rather crucial.

Regarding the second part:
> there is no guarantee that they will evaluate to a string of any length

Isn't this true of any interpolation? What makes interpolations in the middle of a string special?

Based on my interpretation of the code, this condition seems to be about filtering out redundant substitutions. This would happen in the case of adjacent substitutions. And unless I'm mistaken, I think that non-beginning and non-ending empty string literal sections only occur if and only if there's a pair of two adjacent substitutions.

Reply:
> [...] non-beginning and non-ending empty string literal sections only occur if and only if there's a pair of two adjacent substitutions

You're correct, empty strings in the middle of the array would only occur with adjacent substitutions.

> Isn't this true of any interpolation? What makes interpolations in the middle of a string special?

This is true of any interpolation.

The substitution is significant, I meant to convey that the resulting empty quasis are insignificant if present in the middle of the array. Not just redundant but completely lacking of information. Filtering them out doesn't affect how we will match translations.

We have to treat these two statements as equivalent, since we don't know what will be in the substitution:
```js
`hello${''}${''}world`
`hello${' to the people '}${'of the '}world`
```
```json
{
  "helloworld": "matches",
  "hello to the people of the world": "matches"
}
```

That being said, regarding the other comments, I think removing the filtering would be the simplest approach.